pkg/controller/elasticsearch/initcontainer/prepare_fs.go

```diff
@@ -126,12 +127,13 @@ func NewPrepareFSInitContainer(
 			PluginVolumes.InitContainerVolumeMounts(),
 			certificatesVolumeMount,
 			scriptsVolume.VolumeMount(),
-			esvolume.DefaultDataVolumeMount,
 			esvolume.DefaultLogsVolumeMount,
 		),
 		Resources: defaultResources,
 	}
 
+	container.VolumeMounts = esvolume.AppendDefaultDataVolumeMount(container.VolumeMounts, volumes)
```

Comment:
I'm wondering if it is needed since init containers eventually inherit volume mounts from the "main" container: 
https://github.com/elastic/cloud-on-k8s/blob/5915eb125799fc5653d4d2465944f6a8bf382c9d/pkg/controller/common/defaults/pod_template.go#L179-L200

Reply:
I think you are absolutely right. And I believe the same is true for the scripts volume mount and the logs volume mount.